include/yaramod/types/expression_type.h

```diff
@@ -0,0 +1,23 @@
+/**
+ * @file src/types/expression_type.h
+ * @brief Declaration of class ExpressionType.
+ * @copyright (c) 2017 Avast Software, licensed under the MIT license
+ */
+
+#pragma once
+
+namespace yaramod {
+
+///< Type of the expression.
+enum class ExpressionType
```

Comment:
What was the motivation for taking out this type out of `Expression` itself?

Reply:
The `Literal` class now needs to access `Symbol::getName` method. That means it needs to include the _symbol.h_ file. In the `symbol.h` file there has been include of `expression.h` so we would end up with cyclic dependency (see picture). Luckily, the `Symbol` class only needs access to `ExpressionType` which has been extracted from `Expression::Type` to avoid this cyclic dependency.
![yaramod_class_dependecy](https://user-images.githubusercontent.com/29119229/80577689-98babb00-8a07-11ea-91e1-ffeb13c395f1.jpeg)